modules/clientBinaries/generateVersions.js

```diff
@@ -0,0 +1,202 @@
+  var filter = new RegExp(`.*${name}`);
+  sha256 = hashes[platform].match(filter);
+  if (sha256) {
+    sha256 = sha256[0].trim().split(" ")[0];
+  } else {
+    sha256 = undefined;
+  }
+  return (sha256);
+}
+
+/*
+ * get assets for specific platforms
+ */
+var getWinAsset = function(data, asset, hashes) {
+  data.platform = "win";
+  data.arch = asset.name.includes("win64") ? "x64" : "ia32";
+  data.type = asset.content_type === "application/zip" ? "zip" : undefined;
+  data.sha256 = getHash(data.platform, asset.name, hashes);
+}
+
+var getOSXAsset = function (data, asset, hashes) {
+  data.platform = "mac";
+  data.arch = asset.name.includes("osx64") ? "x64" : "ia32";
+  if (asset.content_type === "application/x-apple-diskimage") {
+    data.type = "dmg";
+  } else if (asset.content_type === "application/gzip") {
+    data.type = "tar";
+  }
+  data.sha256 = getHash("osx", asset.name, hashes);
+}
+
+var getLinuxAsset = function (data, asset, hashes) {
+  data.platform = "linux";
+  if (asset.name.includes("x86_64")) {
+    data.arch = "x64";
+  } else if (asset.name.includes("i686")) {
+    data.arch = "ia32";
+  } else if (asset.name.includes("arm")) {
+    data.arch = "arm";
+  }
+  data.type = asset.content_type === "application/gzip" ? "tar" : undefined;
+  data.sha256 = getHash(data.platform, asset.name, hashes);
+}
+
+/*
+ * Gets one asset's details (platform, arch, type. sha256...)
+ */
+var getAssetDetails = function (asset, hashes, version) {
+
+  var data = {
+    platform: undefined,
+    arch: undefined,
+    type: undefined,
+    sha256: undefined
+  };
+
+  // windows binaries
+  if (asset.name.includes("win")) {
+    getWinAsset(data, asset, hashes);
+  } // osx binaries
+  else if (asset.name.includes("osx")) {
+    getOSXAsset(data, asset, hashes);
+  } // linux binaries
+  else if (asset.name.includes("linux")) {
+    getLinuxAsset(data, asset, hashes);
+  }
+
+  // add .exe extension for windows binaries
+  var bin = `particld${data.platform === 'win' ? '.exe' : ''}`
+  // return asset only if it is fully compliant
+  return (data.platform && data.arch && data.type ? {
+    platform: data.platform,
+    arch: data.arch,
+    name: asset.name,
+    entry: {
+      download: {
+        url: asset.browser_download_url,
+        type: data.type,
+        sha256: data.sha256,
+        bin: `particl-${version}/bin/${bin}`
+      },
+      bin: bin,
+      commands: {
+        sanity: {
+          args: ["-version"],
+          output: ["Particl Core Daemon", version]
+        }
+      }
+    }
+  } : undefined);
+}
+
+/*
+ * Gets all hashes of current version for a specific platform
+ */
+var getHashesForPlatform = function (platform, path, hashes, promises) {
+  // this promise is resolved when both HTTP calls returned
+  return new Promise((resolve, reject) => {
+
+    got(`${signaturesURL}/${path}/${maintainer}`).then(response => {
+
+      var files = JSON.parse(response.body);
+      for (id in files) {
```

Comment:
I suggest using
files.forEach(file => {
})
its allot more efficient

Reply:
you are way right